core/domain/feedback_services.py

```diff
@@ -167,7 +167,7 @@ def create_message(
     msg.put()
 
     # Update the message count in the thread.
-    if thread.message_count:
+    if thread.message_count is not None:
         thread.message_count += 1
     else:
         thread.message_count = (
```

Comment:
I believe the issue is here. We don't increment after the message_count = 1 (ie. 2 messages saved). 
```suggestion
        thread.message_count = (
            feedback_models.GeneralFeedbackMessageModel.get_message_count(
                thread_id) + 1)
```

This seems to fix the issue. This was reproducible in the creator interface for feedbacks as well. It was throwing an error when there were more than 2 replies to a feedback thread.

Reply:
@kevintab95 I'm not sure why we have the `else` block in the first place, why do we expect the `message_count` to be None? Also, the changes I've made fixes the issue and this was how it was implemented in the first place but it got changed [here](https://github.com/oppia/oppia/pull/6978/files#diff-e8e75b25b618133593df5ff9cd67a59bR170).

Reply:
Hmmm, so message_count was implemented in a slightly messy fashion. There were some models which might have had it as null. So we ran an audit, and populated all the values, and after that, we removed cases where cases where it would have been null.

Reply:
@DubeySandeep -- `message_count` is 0 when there is a single message (weird) so the `if` condition is to increment to 1, any further addition to messages will go to the else block and it is supposed to increment that value (we do a `thread.put()` later to save the updated values).